### PR TITLE
Implementation of a quantile PDF interpolation Procedure for E-Disps

### PR DESCRIPTION
**Pull request by RuneDominik**
Implements the Quantile PDF Interpolation proposed by @moralejo from [this paper](https://engineering.ucsc.edu/sites/default/files/technical-reports/UCSC-SOE-13-13.pdf) for the interpolation of energy dispersion with one hidden variable. Adresses #161

### CODE REVIEW

**Comment by RuneDominik** [comment]
The latest commit implements the usage of the method for 2D grids by chaining 1D interpolations on a grid decomposed into triangular cells by a Delaunay transformation. This chaining might however propagate interpolation errors in an unpredictable manner and should therefore be tested carefully. The interpolation error does, not surprisingly, rise with the distance between target- and grid-points, as can be seen in the quick test I've done below.

[2D_reference_interpolation.pdf](https://github.com/cta-observatory/pyirf/files/7462993/2D_reference_interpolation.pdf)

File: pyirf/interpolation.py [hunk 1]
```diff
@@ -2,22 +2,239 @@
...
+    cdf = np.cumsum(pdf)
+    interp = interp1d(
+        cdf / cdf.max(), mids, fill_value=(mids.min(), mids.max()), bounds_error=False
+    )
+
+    # Lookup ppf value
+    return interp(percentile)
+
+
+def rebin(entries, mids, width):
```
Comment (adonath) [inline pyirf/interpolation.py:82]: An elegant solution for this problem is to use https://numpy.org/doc/stable/reference/generated/numpy.ufunc.reduceat.html, this way you get ND support and can avoid the Python for loop. However for rebinning along a single axis, the Python for loop might be performant enough. You can use it with arbitrary bins size, just defined by the bin indices.

File: pyirf/interpolation.py [hunk 2]
```diff
@@ -2,22 +2,239 @@
...
+    # positions q_bar.
+    width = np.diff(edges)
+    helper = np.concatenate((q_bar, V_bar), axis=axis)
+    interpolated_histogram = np.apply_along_axis(rebin, axis, helper, mids, width)
+
+    # Re-Normalize, as the normalisation is lost due to approximate nature of this method
+    # Set norm to nan for empty histograms to avoid division through 0
+    if normalize == "sum":
+        norm = np.sum(interpolated_histogram, axis=axis)
+        norm = np.repeat(
```
Comment (adonath) [inline pyirf/interpolation.py:214]: I think there is now need to repeat the values here, you can just rely on numpy broadcasting.

**Comment by RuneDominik** [comment]
The #174 version should be sufficient for now and is much leaner. If we find that this "original" method has advantages we can still revisit this.

**Comment by maxnoe** [comment]
Ok, great, then I'll go ahead and make the release

**Pull request closed without merging**